### ISSUE TITLE
Implement more intelligent way of defining FIR filter length

### ISSUE THREAD

**Issue by psurukuc** [project8/locust_mc#128]
The current FIR filter length (while generating from TF) is done by taking 2*(circular shift) + (TF length). This works alright for most of the filters, but if the filter is really resonant it fails. Implement a better way to do this by perhaps defining both the ends of the filter where the magnitude falls below a threshold (as a fraction of the maximum magnitude of the filter ?).

**Comment by arinabykadorova** [comment]
Yes, this would be better. Perhaps a patch-up longer FIR length is okay for now, but will put this on the list.

**Comment by pslocum** [comment]
@jkgaison65 has been able to adapt the antenna FIRs for resonant systems.  So I think we can optimize the FIRs for antennas, merge to develop, and then revisit if needed when we merge the resonant calculations.  In the meantime if there are additional near-term FIR changes being planned, please let us know.
